inst/extdata/install-packages.R

```diff
@@ -0,0 +1,163 @@
+  # returns TRUE if PK-Sim.R dll can be loaded,
+  # or raises errors otherwise
+  library(ospsuite)
+  ospsuite::initPKSim(pkSimPath)
+  return(TRUE)
+}
+
+# Remove functions created by this script from the environment
+cleanEnvironment <- function(){
+  rm(packageInstallationMessages,
+          testPKSIMConnection,
+          cleanEnvironment,
+          displayProgress,
+          installOSPPackages,
+          installPackagesGlobally
+  )
+}
+
+displayProgress <- function(current, success = TRUE, message = NULL, suppressOutput = TRUE) {
+  states <- c("Installing RENV", "Installing CRAN packages",
+              "Checking RTOOLS", "Installing rClr", "Installing ospsuite.utils",
+              "Installing tlf", "Installing ospsuite",
+              "Testing PK-Sim connection",
+              "Installation successful")
+  if (suppressOutput) {
+    cat("\014")
+    for (i in seq_along(states)) {
+      cat(paste0(ifelse(i < which(states == current), "V", ifelse(i > which(states == current), ".", ifelse(success == FALSE, "X", ".")))), " ", states[[i]], "\n")
+    }
+  } else {
+    cat(paste0(current, "\n"))
+  }
+  if (!is.null(message)) {
+    message(message)
+  }
+}
+
+#' Install all osps packages and their dependencies.
+#'
+#' @param rtoolsPath Path to where rtools are installed. If `NULL` (default),
+#' the path is deduced from system environment variables.
+#' @param rclrVersion Version of rClr package. Default is 0.9.2 for Windows R4.
+#' @param developerVersion If `FALSE` (default), release verions of the packages
+#' will be installed. If `TRUE`, latest developer builds of the osps packages
+#' will be installed
+#' @param suppressOutput
+#'
+#' @return
+#' @export
+#'
+#' @examples
+installOSPPackages <- function(rtoolsPath = NULL, rclrVersion = "0.9.2",
+                               suppressOutput = TRUE,
+                               developerVersion = FALSE){
+  # Install dependencies from CRAN
+  displayProgress("Installing CRAN packages", suppressOutput = suppressOutput)
+  install.packages(c("R6", "stringr", "readr", "hash", "readxl", "shiny", "shinyjs", "vctrs", "writexl", "dplyr", "tidyr", "ggplot2", "FME", "patchwork"),
+                   dependencies = TRUE)
+
+  displayProgress("Checking RTOOLS", suppressOutput = suppressOutput)
+  if (Sys.which("make") == "") { # rtools is not found
+    if (!is.null(rtoolsPath)) { # adding an existing installation of rtools to path
+      Sys.setenv(PATH = paste(rtoolsPath, Sys.getenv("PATH"), sep = ";"))
+    } else {
+      displayProgress("Checking RTOOLS", success = FALSE, message = packageInstallationMessages$installRTools, suppressOutput = suppressOutput)
+      return()
+    }
+  }
+  if (Sys.which("make") == "") {
+    displayProgress("Checking RTOOLS", success = FALSE, message = packageInstallationMessages$RToolsNotFound, suppressOutput = suppressOutput)
+    return()
+  }
+
+  displayProgress("Installing rClr", suppressOutput = suppressOutput)
```

Comment:
Install rClr - currently only for Windows

Reply:
This must be extendened for all currently supported systems (Ubuntu18, CentOS7)